src/rules/no-deprecated-functions.ts

```diff
@@ -22,13 +78,27 @@ export default createRule({
   },
   defaultOptions: [],
   create(context) {
+    const jestVersion =
+      (context.settings as ContextSettings)?.jest?.version ||
```

Comment:
Can we do something so we don't need the type cast?

Reply:
No, as `context.settings` is just type `object` - it ideally should take a generic, but it's already two that are heavily inferred which means a generic would be unwieldy without support for a sigil to indicate [partial type argumentation inference](https://github.com/Microsoft/TypeScript/pull/26349)